ci/dhis2-verify-lib.yml

```diff
@@ -0,0 +1,140 @@
+    GH_TOKEN: ${{secrets.GH_TOKEN}}
+    CI: true
+
+jobs:
+    install:
+        runs-on: ubuntu-latest
+        if: "!contains(github.event.head_commit.message, '[skip ci]')"
+
+        steps:
+            - uses: actions/checkout@v2
+            - uses: actions/setup-node@v1
+              with:
+                  node-version: 12.x
+
+            - name: Get yarn cache directory path
+              id: yarn-cache-dir-path
+              run: echo "::set-output name=dir::$(yarn cache dir)"
+
+            - uses: actions/cache@v2
+              id: yarn-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
+              with:
+                  path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+                  key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+                  restore-keys: |
+                      ${{ runner.os }}-yarn-
+
+            - name: Install
+              run: yarn install --frozen-lockfile
+
+    lint:
+        runs-on: ubuntu-latest
+        needs: install
+        if: "!contains(github.event.head_commit.message, '[skip ci]')"
+        steps:
+            - uses: actions/checkout@v2
+            - uses: actions/setup-node@v1
+              with:
+                  node-version: 12.x
+
+            - name: Get yarn cache directory path
+              id: yarn-cache-dir-path
+              run: echo "::set-output name=dir::$(yarn cache dir)"
+
+            - uses: actions/cache@v2
+              id: yarn-cache
+              with:
+                  path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+                  key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+                  restore-keys: |
+                      ${{ runner.os }}-yarn-
+
+            - name: Install
+              run: yarn install --frozen-lockfile
+
+            - name: Lint
+              run: |
+                  ./bin/d2-style js check
+                  ./bin/d2-style text check
+
+    test:
+        runs-on: ubuntu-latest
+        needs: install
+        if: "!contains(github.event.head_commit.message, '[skip ci]')"
+        steps:
+            - uses: actions/checkout@v2
+            - uses: actions/setup-node@v1
+              with:
+                  node-version: 12.x
+
+            - name: Get yarn cache directory path
+              id: yarn-cache-dir-path
+              run: echo "::set-output name=dir::$(yarn cache dir)"
+
+            - uses: actions/cache@v2
+              id: yarn-cache
+              with:
+                  path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+                  key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+                  restore-keys: |
+                      ${{ runner.os }}-yarn-
+
+            - name: Install
+              run: yarn install --frozen-lockfile
+
+            - name: Test
+              run: yarn test
+
+    publish:
+        runs-on: ubuntu-latest
+        needs: [install, lint, test]
```

Comment:
Libs also need to be build before publication, I believe?  Looks like the lib workflow doesn't have a build step

Reply:
Yep! Thanks. I was using cli-style as a test bed and that doesn't need a build step so that slipped out of my brain. Should we have a specific one for cli in addition to lib and app?

Reply:
Made a new one, `dhis2-verify-node.yml`.